File: Js_solution/easy/Solution2373.js
/**
 * @param {number[][]} grid
 * @return {number[][]}
 */
var largestLocal = function(grid) {
    let result = Array(grid.length-2).fill(0).map(()=>Array(grid.length-2));
    for (let i=0;i<result.length;i++){
        for (let j=0;j<result[0].length;j++){
            result[i][j] = findTheMaxNumber(i,j,grid);
        }
    }
    return result;
};

var findTheMaxNumber = function(i,j,grid){
    var manNumber = 0;
    for (let k=i;k<=i+2;k++){
        for (let m = j;m<=j+2;m++){
            manNumber = Math.max(manNumber,grid[k][m]);
        }
    }
    return manNumber;
}

File: python_solution/easy/Solution2373.py
from typing import List


class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        list_result = [[0 for x in range(len(grid)-2)] for y in range(len(grid)-2)]
        for i in range(len(list_result)):
            for j in range(len(list_result[0])):
                list_result[i][j] = self.findTheMaxNumber(i,j,grid)
        return list_result

    def findTheMaxNumber(self,i:int,j:int,grid:List[List[int]])->int:
        maxNumber = 0
        for k in range(i,i+3):
            for m in range(j,j+3):
                maxNumber = max(maxNumber,grid[k][m])
        return maxNumber